File: year2/mpp/java/MotorcycleContest/src/main/java/domain/validators/ParticipantValidator.java
package domain.validators;

import domain.ValidationException;
import domain.entities.Participant;
import org.springframework.stereotype.Component;

@Component
public class ParticipantValidator implements Validator<Participant> {
    @Override
    public void validate(Participant participant) throws ValidationException {
        if (participant.getName() == null || participant.getRaceId() == null || participant.getTeamId() == null) {
            throw new ValidationException("All participant fields must be not null");
        }
        if (participant.getName().equals("")) {
            throw new ValidationException("All participant fields must be not empty");
        }
        if (participant.getTeamId() < 0 || participant.getRaceId() < 0) {
            throw new ValidationException("Participant ids must be positive integers");
        }
    }
}

File: year2/mpp/c#/MotorcycleClientServer/MotorcycleCommon/Validators/ParticipantValidator.cs
using MotorcycleContest.Domain.Entities;

namespace MotorcycleContest.Domain.Validators
{
    public class ParticipantValidator : IValidator<Participant>
    {
        public void Validate(Participant participant)
        {
            if (participant.Name.Equals(null) || participant.RaceId.Equals(-1) || participant.TeamId.Equals(-1))
            {
                throw new ValidationException("null fields");
            }
            if (participant.Name.Equals(""))
            {
                throw new ValidationException("name is empty");
            }
            if (participant.TeamId < 0 || participant.RaceId < 0)
            {
                throw new ValidationException("ids are negative integers");
            }
        }
    }
}
